Explain what "lazy" means in this context

Can you update the readme to explain at what point the module is loaded when loaded via "lazy"?

I try to load module lazy. But `webpack` give me this:

```bash
WARNING in ./src/main.js
Critical dependencies:
47:28-78 the request of a dependency is an expression
 @ ./src/main.js 47:28-78

WARNING in ./src/index.html
Module parse failed: /Users/dulin/workspace/webpack-summer/bundle-loader/simple/src/index.html Unexpected token (1:0)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (1:0)
    at Parser.pp.raise (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:920:13)
    at Parser.pp.unexpected (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:1483:8)
    at Parser.pp.parseExprAtom (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:330:12)
    at Parser.pp.parseExprSubscripts (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:225:19)
    at Parser.pp.parseMaybeUnary (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:204:17)
    at Parser.pp.parseExprOps (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:151:19)
    at Parser.pp.parseMaybeConditional (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:133:19)
    at Parser.pp.parseMaybeAssign (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:110:19)
    at Parser.pp.parseExpression (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:86:19)
    at Parser.pp.parseStatement (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:1753:23)
    at Parser.pp.parseTopLevel (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:1648:21)
    at Parser.parse (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:1616:17)
    at Object.parse (/usr/local/lib/node_modules/webpack/node_modules/acorn/dist/acorn.js:882:44)
    at Parser.parse (/usr/local/lib/node_modules/webpack/lib/Parser.js:902:15)
    at DependenciesBlock.<anonymous> (/usr/local/lib/node_modules/webpack/lib/NormalModule.js:104:16)
    at DependenciesBlock.onModuleBuild (/usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:310:10)
    at nextLoader (/usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:275:25)
    at /usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.provide (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:52:20)
    at CachedInputFileSystem.readFile (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:140:24)
    at DependenciesBlock.onLoadPitchDone (/usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:255:7)
    at DependenciesBlock.loadPitch (/usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:182:27)
    at DependenciesBlock.doBuild (/usr/local/lib/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:241:4)
    at DependenciesBlock.build (/usr/local/lib/node_modules/webpack/lib/NormalModule.js:84:14)
    at Compilation.buildModule (/usr/local/lib/node_modules/webpack/lib/Compilation.js:126:9)
    at /usr/local/lib/node_modules/webpack/lib/Compilation.js:309:10
    at /usr/local/lib/node_modules/webpack/lib/NormalModuleFactory.js:58:13
    at NormalModuleFactory.applyPluginsAsyncWaterfall (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/Tapable.js:75:69)
    at onDoneResolving (/usr/local/lib/node_modules/webpack/lib/NormalModuleFactory.js:38:11)
    at onDoneResolving (/usr/local/lib/node_modules/webpack/lib/NormalModuleFactory.js:121:6)
 @ ./src ^\.\/.*$

WARNING in ./src ^\.\/.*$
Module not found: Error: a dependency to an entry point is not allowed
 @ ./src ^\.\/.*$

WARNING in ./src ^\.\/.*$
Module not found: Error: a dependency to an entry point is not allowed
 @ ./src ^\.\/.*$
```

here is my code:

```js
document.addEventListener('DOMContentLoaded', init);

const dynamicModules = ['a', 'b', 'c'];

function init() {
	document.getElementById('lazyloadModule').addEventListener('click', handleLazyloadModule, false);
}

function handleLazyloadModule(e) {
	var moduleName = 'a';
	var layzLoadModuleMap = loadModuleLazy(dynamicModules);
	layzLoadModuleMap[moduleName](function(module) {
		console.log('lazyloadModule is ' + module);
	})
}

function loadModuleLazy(names) {
	var layzLoadModuleMap = {};
	names.forEach(function(name) {
		layzLoadModuleMap[name] = require(`bundle-loader?lazy!./modules/${name}.js`);
	});
	return layzLoadModuleMap;
}
```

Anybody can help me? Thanks.

--- a/README.md
+++ b/README.md
@@ -1,15 +1,17 @@
 [![npm][npm]][npm-url]
+[![node][node]][node-url]
 [![deps][deps]][deps-url]
+[![tests][tests]][tests-url]
+[![coverage][cover]][cover-url]
 [![chat][chat]][chat-url]
 
 <div align="center">
-  <!-- replace with accurate logo e.g from https://worldvectorlogo.com/ -->
   <a href="https://github.com/webpack/webpack">
-    <img width="200" height="200" vspace="" hspace="25"
+    <img width="200" height="200"
       src="https://cdn.rawgit.com/webpack/media/e7485eb2/logo/icon.svg">
   </a>
   <h1>Bundle Loader</h1>
-  <p>Bundle loader for webpack.<p>
+  <p>Bundle loader for webpack<p>
 </div>
 
 <h2 align="center">Install</h2>
@@ -20,65 +22,134 @@ npm i bundle-loader --save
 
 <h2 align="center"><a href="https://webpack.js.org/concepts/loaders">Usage</a></h2>
 
-``` javascript
-// The chunk is requested, when you require the bundle
-var waitForChunk = require("bundle-loader!./file.js");
-
-// To wait until the chunk is available (and get the exports)
-//  you need to async wait for it.
-waitForChunk(function(file) {
-	// use file like it was required with
-	// var file = require("./file.js");
-});
-// wraps the require in a require.ensure block
-
-// Multiple callbacks can be added. They will be executed in the order of addition. 
-waitForChunk(callbackTwo);
-waitForChunk(callbackThree);
-// If a callback is added after dependencies were loaded, it will be called immediately.
-```
-
-The file is requested when you require the bundle loader. If you want it to request it lazy, use:
-
-``` javascript
-var load = require("bundle-loader?lazy!./file.js");
-
-// The chunk is not requested until you call the load function
-load(function(file) {
-
-});
-```
-### `name` query parameter
-
-You may set name for a bundle using the `name` query parameter. 
-See [documentation](https://github.com/webpack/loader-utils#interpolatename).
-
-**Note** chunks created by the loader will be named according to the 
-[`output.chunkFilename`](https://webpack.js.org/configuration/output/#output-chunkfilename) rule, which defaults to `[id].[name]`.
-Here `[name]` corresponds to the chunk name set in the `name` query parameter. 
-
-#### Example:
-
-``` js
-require("bundle-loader?lazy&name=my-chunk!./file.js");
-require("bundle-loader?lazy&name=[name]!./file.js");
-```
-And the webpack configuration:
-``` js
+**webpack.config.js**
+```js
 module.exports = {
-   entry: { ... },
-   output : {
-      path : ...,
-      filename : '[name].js',
-      chunkFilename : '[name]-[id].js', // or whatever other format you want.
-   },
+  module: {
+    rules: [
+      {
+        test: /\.bundle\.js$/,
+        use: 'bundle-loader'
+      }
+    ]
+  }
 }
 ```
 
-Normal chunks will show up using the `filename` rule above, and be named according to their chunkname. 
-Chunks from `bundle-loader`, however will load using the `chunkFilename` rule, so the example files will produce `my-chunk-1.js` and `file-2.js` respectively.
+The chunk is requested, when you require the bundle.
 
-You can also use `chunkFilename` to add hash values to the filename, since putting `[hash]` in the bundle query parameter does not work correctly. 
+**file.js**
+```js
+import bundle from './file.bundle.js';
+```
+
+To wait until the chunk is available (and get the exports)
+you need to async wait for it.
+
+```js
+bundle((file) => {
+  // use the file like it was required
+  const file = require('./file.js')
+});
+```
+
+This wraps the `require('file.js')` in a `require.ensure` block
+
+Multiple callbacks can be added. They will be executed in the order of addition.
+
+```js
+bundle(callbackTwo)
+bundle(callbackThree)
+```
+
+If a callback is added after dependencies were loaded, it will be called immediately.
+
+<h2 align="center">Options</h2>
+
+|Name|Type|Default|Description|
+|:--:|:--:|:-----:|:----------|
+|**`lazy`**|`{Boolean}`|`false`|Loads the imported bundle asynchronously|
+|**`name`**|`{String}`|`[id].[name]`|Configure a custom filename for your imported bundle|
+
+### `lazy`
+
+The file is requested when you require the `bundle-loader`. If you want it to request it lazy, use:
+
+**webpack.config.js**
+```js
+{
+  loader: 'bundle-loader',
+  options: {
+    lazy: true
+  }
+}
+```
+
+```js
+import bundle from './file.bundle.js'
+
+bundle((file) => {...})
+```
+
+> ℹ️  The chunk is not requested until you call the load function
+
+### `name`
+
+You may set name for a bundle using the `name` options parameter.
+See [documentation](https://github.com/webpack/loader-utils#interpolatename).
+
+**webpack.config.js**
+```js
+{
+  loader: 'bundle-loader',
+  options: {
+    name: '[name]'
+  }
+}
+```
+
+> :warning: chunks created by the loader will be named according to the
+[`output.chunkFilename`](https://webpack.js.org/configuration/output/#output-chunkfilename) rule, which defaults to `[id].[name]`. Here `[name]` corresponds to the chunk name set in the `name` options parameter.
+
+<h2 align="center">Examples</h2>
+
+```js
+import bundle from './file.bundle.js'
+```
+
+**webpack.config.js**
+``` js
+module.exports = {
+  entry: {
+   index: './App.js'
+  },
+  output: {
+    path: path.resolve(__dirname, 'dest'),
+    filename: '[name].js',
+    // or whatever other format you want
+    chunkFilename: '[name].[id].js',
+  },
+  module: {
+    rules: [
+      {
+        test: /\.bundle\.js$/,
+        use: {
+          loader: 'bundle-loader'
+          options: {
+            name: 'my-chunk'
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+Normal chunks will show up using the `filename` rule above, and be named according to their `[chunkname]`.
+
+Chunks from `bundle-loader`, however will load using the `chunkFilename` rule, so the example files will produce `my-chunk.1.js` and `file-2.js` respectively.
+
+You can also use `chunkFilename` to add hash values to the filename, since putting `[hash]` in the bundle options parameter does not work correctly.
 
 <h2 align="center">Maintainers</h2>
 
@@ -86,28 +157,32 @@ You can also use `chunkFilename` to add hash values to the filename, since putti
   <tbody>
     <tr>
       <td align="center">
-        <img width="150" height="150"
-        src="https://avatars3.githubusercontent.com/u/166921?v=3&s=150">
-        </br>
-        <a href="https://github.com/bebraw">Juho Vepsäläinen</a>
+        <a href="https://github.com/bebraw">
+          <img width="150" height="150" src="https://github.com/bebraw.png?v=3&s=150">
+          </br>
+          Juho Vepsäläinen
+        </a>
       </td>
       <td align="center">
-        <img width="150" height="150"
-        src="https://avatars2.githubusercontent.com/u/8420490?v=3&s=150">
-        </br>
-        <a href="https://github.com/d3viant0ne">Joshua Wiens</a>
+        <a href="https://github.com/d3viant0ne">
+          <img width="150" height="150" src="https://github.com/d3viant0ne.png?v=3&s=150">
+          </br>
+          Joshua Wiens
+        </a>
       </td>
       <td align="center">
-        <img width="150" height="150"
-        src="https://avatars3.githubusercontent.com/u/533616?v=3&s=150">
-        </br>
-        <a href="https://github.com/SpaceK33z">Kees Kluskens</a>
+        <a href="https://github.com/michael-ciniawsky">
+          <img width="150" height="150" src="https://github.com/michael-ciniawsky.png?v=3&s=150">
+          </br>
+          Michael Ciniawsky
+        </a>
       </td>
       <td align="center">
-        <img width="150" height="150"
-        src="https://avatars3.githubusercontent.com/u/3408176?v=3&s=150">
-        </br>
-        <a href="https://github.com/TheLarkInn">Sean Larkin</a>
+        <a href="https://github.com/evilebottnawi">
+          <img width="150" height="150" src="https://github.com/evilebottnawi.png?v=3&s=150">
+          </br>
+          Alexander Krasnoyarov
+        </a>
       </td>
     </tr>
   <tbody>
@@ -117,8 +192,17 @@ You can also use `chunkFilename` to add hash values to the filename, since putti
 [npm]: https://img.shields.io/npm/v/bundle-loader.svg
 [npm-url]: https://npmjs.com/package/bundle-loader
 
+[node]: https://img.shields.io/node/v/bundle-loader.svg
+[node-url]: https://nodejs.org
+
 [deps]: https://david-dm.org/webpack-contrib/bundle-loader.svg
 [deps-url]: https://david-dm.org/webpack-contrib/bundle-loader
 
-[chat]: https://img.shields.io/badge/gitter-webpack%2Fwebpack-brightgreen.svg
+[tests]: http://img.shields.io/travis/webpack-contrib/bundle-loader.svg
+[tests-url]: https://travis-ci.org/webpack-contrib/bundle-loader
+
+[cover]: https://coveralls.io/repos/github/webpack-contrib/bundle-loader/badge.svg
+[cover-url]: https://coveralls.io/github/webpack-contrib/bundle-loader
+
+[chat]: https://badges.gitter.im/webpack/webpack.svg
 [chat-url]: https://gitter.im/webpack/webpack

